File: ui/src/lib/time.ts
export const convertZulu2Beijing = (zuluTime: string) => {
  const utcDate = new Date(zuluTime);

  // Format the Beijing time
  const formattedBeijingTime = new Intl.DateTimeFormat("zh-CN", {
    year: "numeric",
    month: "2-digit",
    day: "2-digit",
    hour: "2-digit",
    minute: "2-digit",
    second: "2-digit",
    hour12: false,
    timeZone: "Asia/Shanghai",
  }).format(utcDate);

  return formattedBeijingTime;
};

export const getLeftDays = (zuluTime: string) => {
  const time = convertZulu2Beijing(zuluTime);
  const date = time.split(" ")[0];
  const now = new Date();
  const target = new Date(date);
  const diff = target.getTime() - now.getTime();
  const days = Math.floor(diff / (1000 * 60 * 60 * 24));
  return days;
};

export const diffDays = (date1: string, date2: string) => {
  const target1 = new Date(date1);
  const target2 = new Date(date2);
  const diff = target1.getTime() - target2.getTime();
  const days = Math.ceil(diff / (1000 * 60 * 60 * 24));
  return days;
};

File: internal/utils/xtime/time.go
package xtime

import (
	"time"
)

func BeijingTimeStr() string {
	location, _ := time.LoadLocation("Asia/Shanghai")

	// 获取当前时间
	now := time.Now().In(location)

	// 格式化为字符串
	return now.Format("2006-01-02 15:04:05")
}

func GetTimeAfter(d time.Duration) string {

	t := time.Now().UTC()

	return t.Add(d).Format("2006-01-02 15:04:05")
}
